### PR TITLE
Merge privacy manifests and copy to build output

### LINKED ISSUE #354: Copy Apple privacy manifest files to build output
We need to start sending back the Apple privacy manifests of resolved pods to the build client.

### PR DESCRIPTION
**Pull request by britzl**
This change will gather all `*.xcprivacy` files from upload directory and from resolved pods and merge them into a single `PrivacyInfo.xcprivacy` and send the merged file together with other output files to the client.

Fixes #354

### CODE REVIEW

File: server/manifestmergetool/src/main/java/com/defold/manifestmergetool/AndroidManifestMerger.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package com.defold.manifestmergetool;
```
Comment (britzl) [inline server/manifestmergetool/src/main/java/com/defold/manifestmergetool/AndroidManifestMerger.java:1]: Move this to a separate file. We have a separate merge class for HTML and plists so why not one AndroidManifest.xml?

File: server/manifestmergetool/src/main/java/com/defold/manifestmergetool/ManifestMergeTool.java [hunk 2]
```diff
@@ -9,47 +9,20 @@
...
-import com.android.manifmerger.XmlDocument;
-import com.android.utils.ILogger;
-
 public class ManifestMergeTool {
 
     public enum Platform {
         ANDROID, IOS, OSX, WEB, UNKNOWN
     }
 
-    private static class ILoggerWrapper implements ILogger {
```
Comment (britzl) [inline server/manifestmergetool/src/main/java/com/defold/manifestmergetool/ManifestMergeTool.java:32]: Moved to AndroidManifestMerger

File: server/manifestmergetool/src/main/java/com/defold/manifestmergetool/ManifestMergeTool.java [hunk 3]
```diff
@@ -59,44 +32,26 @@ public void usage() {
     }
 
     private static void mergeAndroid(File main, File[] libraries, File out) throws RuntimeException {
-        // Good explanation of merge rules: https://android.googlesource.com/platform/tools/base/+/jb-mr2-dev/manifest-merger/src/main/java/com/android/manifmerger/ManifestMerger.java
```
Comment (britzl) [inline server/manifestmergetool/src/main/java/com/defold/manifestmergetool/ManifestMergeTool.java:62]: Moved to AndroidManifestMerger

File: server/manifestmergetool/src/main/java/com/defold/manifestmergetool/ManifestMergeTool.java [hunk 4]
```diff
@@ -59,44 +32,26 @@ public void usage() {
...
+        AndroidManifestMerger merger = new AndroidManifestMerger(logger);
+        merger.merge(main, libraries, out);
     }
 
     private static void mergePlist(File main, File[] libraries, File out) throws RuntimeException {
         InfoPlistMerger merger = new InfoPlistMerger(logger);
         merger.merge(main, libraries, out);
     }
 
+    private static void mergeApple(File main, File[] libraries, File out) throws RuntimeException {
```
Comment (britzl) [inline server/manifestmergetool/src/main/java/com/defold/manifestmergetool/ManifestMergeTool.java:44]: Send .xcprivacy files to a bespoke merger. Use InfoPlistMerger for everything else.

File: server/src/main/java/com/defold/extender/Extender.java [hunk 5]
```diff
@@ -2370,6 +2370,48 @@ private List<File> buildAndroid(String platform) throws ExtenderException {
         return outputFiles;
     }
 
+    private List<File> buildApple(String platform) throws ExtenderException {
```
Comment (britzl) [inline server/src/main/java/com/defold/extender/Extender.java:2373]: Merge xcprivacy files

**Review by JCash — APPROVED** [review]

**Pull request merged**